music21/musicxml/xmlToM21.py

```diff
@@ -867,6 +867,21 @@ def xmlRootToScore(self, mxScore, inputM21=None):
             self.spannerBundle.remove(sp)
 
         s.coreElementsChanged()
+        # Fill gaps with rests where needed
+        for m in s[stream.Measure]:
```

Comment:
Why move this out of the MeasureParser? If it's just to have access to the `md` instance, we could just query for it in the measure parser (with getContextByClass, which is fast). To do that, we likely need to avoid `coreInsert`'ing it when inserting for the reason [mentioned](https://github.com/cuthbertLab/music21/pull/1636#issuecomment-2143924582) earlier.

Reply:
I had a look - I moved it there since it didn't play nice with partwise musicxml files. I think this is because the rests inserted by `makeRests` don't have an entry in `staffReference`, and therefore can't be placed in the correct part until after the parts have been pulled apart. 

One example is schoenberg op19.2. - `testHiddenRests` fails when keeping this inside the measure parser because a hole in voice 2 is filled with a half rest. this voice is supposed to be the top staff, but the rest ends up in the bottom staff, where we end up with two voices instead of no voices (one with the newly inserted rest, and one with the actual data for the bottom staff).

Reply:
I wasn't able to quickly resolve this in another way - lmk if you have any thoughts.

Reply:
I think this test wasn't failing before because we got luckky and this particular case happened in the last measure of the piece, where the incorrect rest was removed by `removeEndForwardRest`.